internal/cmds/setting/argument/fields.go

```diff
@@ -0,0 +1,58 @@
+package argument
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/djdv/go-filesystem-utils/internal/cmds/setting/runtime"
+)
+
+func expandEmbedded(ctx context.Context, fields runtime.SettingsFields) runtime.SettingsFields {
+	out := make(chan reflect.StructField, cap(fields))
+	go func() {
+		defer close(out)
+		for field := range fields {
+			if !field.Anonymous ||
+				field.Type.Kind() != reflect.Struct {
+				select {
+				case out <- field:
+					continue
+				case <-ctx.Done():
+					return
+				}
+			}
+			for field := range expandField(ctx, field) {
+				select {
+				case out <- field:
+				case <-ctx.Done():
+					return
+				}
+			}
+		}
+	}()
+	return out
```

Comment:
good function but maybe out could have a more descriptive name? not that i can think of one 🤡